<!--REQUEST_ID=58b201dd-a0fd-4d3b-8be8-77f2619fe088-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46674&attempt_id=a4c7c8a5-5fd4-48ea-98cc-db1a63ffebe7)

Attempted: mkspiffs, mkspiffs-presets.arduino-esp8266, mkspiffs-presets.arduino-esp32, mkspiffs-presets.esp-idf

<details><summary>Partial log (click to expand)</summary><p>

```
      _spiffs_object_create in spiffs_nucleus.o
      _spiffs_object_update_index_hdr in spiffs_nucleus.o
  "start", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture i386
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:106: mkspiffs] Error 1
builder for '/nix/store/93qih1v1wmkv812zrsfb2yldfk1398f1-mkspiffs-0.2.3.drv' failed with exit code 2
builder for '/nix/store/x8c30s9216yczz90f0jafab0p20j53dj-mkspiffs-0.2.3.drv' failed with exit code 2
error: build of '/nix/store/145akjfyrcd7nbphr8a0vimg44499iz1-mkspiffs-0.2.3.drv', '/nix/store/93qih1v1wmkv812zrsfb2yldfk1398f1-mkspiffs-0.2.3.drv', '/nix/store/qv3k2rbsg737xcgd6pzl3n9v0bwwwfhy-mkspiffs-0.2.3.drv', '/nix/store/x8c30s9216yczz90f0jafab0p20j53dj-mkspiffs-0.2.3.drv' failed
```
</p></details>

--- a/pkgs/tools/filesystems/mkspiffs/default.nix
+++ b/pkgs/tools/filesystems/mkspiffs/default.nix
@@ -1,7 +1,7 @@
 { stdenv, fetchFromGitHub, git }:
 
-# We can build mkspiffs with custom CPPFLAGS by passing them in extraBuidFlags.
-# The upstream has several named presets for CPPFLAGS whose names will be passed in buildConfigName.
+# Changing the variables CPPFLAGS and BUILD_CONFIG_NAME can be done by
+# overriding the same-named attributes. See ./presets.nix for examples.
 
 stdenv.mkDerivation rec {
   name = "mkspiffs-${version}";

--- a/pkgs/tools/filesystems/mkspiffs/default.nix
+++ b/pkgs/tools/filesystems/mkspiffs/default.nix
@@ -27,5 +27,6 @@ stdenv.mkDerivation rec {
     license = licenses.mit;
     homepage = https://github.com/igrr/mkspiffs;
     maintainers = with maintainers; [ haslersn ];
+    platforms = platforms.linux;
   };
 }

--- a/pkgs/tools/filesystems/mkspiffs/presets.nix
+++ b/pkgs/tools/filesystems/mkspiffs/presets.nix
@@ -1,5 +1,7 @@
 { lib, mkspiffs }:
 
+# We provide the same presets as the upstream
+
 lib.mapAttrs (
   name: { CPPFLAGS }:
   mkspiffs.overrideAttrs (drv: {

